### ISSUE TITLE
Bug for Insert record

### ISSUE THREAD

**Issue by martingg88** [orientechnologies/orientdb#5686]
I do have following bug when insert record for version 2.1.8. Any idea?

```
tive. [OServer]{db=voyage} Error on creating record in cluster: plocal cluster: cronjob
com.orientechnologies.orient.core.exception.OStorageException
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.createRecord(OPaginate
dCluster.java:434)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doCreateRecord(OAbstract
PaginatedStorage.java:1963)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.createRecord(OAbstractPa
ginatedStorage.java:786)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeSaveRecord(ODatabaseDocumentTx
.java:1999)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:159)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2568)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:121)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1768)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1759)
        at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:81)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.saveRecord(OCommandExecutorSQLInsert.ja
va:323)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.execute(OCommandExecutorSQLInsert.java:
224)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.j
ava:90)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstract
PaginatedStorage.java:1547)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginat
edStorage.java:1528)
        at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstra
ct.java:67)
        at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostCommand.execut
e(OServerCommandPostCommand.java:81)
        at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetwork
ProtocolHttpAbstract.java:180)
        at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetwork
ProtocolHttpAbstract.java:627)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
Caused by: com.orientechnologies.common.directmemory.ODirectMemoryViolationException: Provided offset [119288] is
more than size of allocated area  [65552]
        at com.orientechnologies.common.directmemory.OUntrackedDirectMemoryPointer.rangeCheck(OUntrackedDirectMemo
ryPointer.java:236)
        at com.orientechnologies.common.directmemory.OUntrackedDirectMemoryPointer.getInt(OUntrackedDirectMemoryPo
inter.java:107)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWALChangesTree.getIntValue(OWALChan
gesTree.java:89)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OWALChangesTree$PointerWrapper.getIn
t(OWALChangesTree.java:783)
        at com.orientechnologies.common.serialization.types.OIntegerSerializer.deserializeFromDirectMemory(OIntege
rSerializer.java:125)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.base.ODurablePage.getIntValue(ODurablePa
ge.java:101)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OClusterPage.appendRecord(OClusterPage.j
ava:103)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.addEntry(OPaginatedClu
ster.java:1482)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedCluster.createRecord(OPaginate
dCluster.java:423)
        ... 19 more

2016-02-04 15:20:53:172 SEVER Internal server error:
java.lang.NullPointerException [ONetworkProtocolHttpDb]
```

**Comment by maggiolo00** [comment]
@laa 
any idea?

**Comment by laa** [comment]
Well it depends how long ago this cluster was created. 
Did you create cluster from scratch or that is cluster and data created in 2.0 version ?

Also is it possible to look at this cluster file ?

**Comment by martingg88** [comment]
This have been created for long time since from 2.0 until now.
i just got this issue suddenly. very faint now. argss.....
what can i do to solve this issue?

**Comment by laa** [comment]
You may send me this cluster and I will fix it . This issue was in 2.0 and in early versions of 2.1 , so probably it is fixed but record left corrupted. Sorry for that.

**Comment by martingg88** [comment]
@laa how and where should i retrieve the cluster file?

**Comment by laa** [comment]
@martingg88 that is set of files which have the same name as class name underneath of database directory. Could you zip all of them and send ?

**Comment by martingg88** [comment]
@laa, We can't share this to you as it consist of customer data. Hope you understand. Do we have another way to work around?

**Comment by laa** [comment]
Yes, but it is time consuming, you need to export all data to json and then
back. But it  is mandatory to export all data because otherwise links
between data will be broken.

On Tue, Feb 9, 2016 at 5:06 PM martingg88 <notifications@github.com> wrote:

> @laa <https://github.com/laa>, We can't share this to you as it consist
> of customer data. Hope you understand. Do we have another way to work
> around?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/orientechnologies/orientdb/issues/5686#issuecomment-181906925>
> .
>
-- 
Best regards,
Andrey Lomakin, R&D lead.
OrientDB Ltd

twitter:@Andrey_Lomakin linkedin:https://ua.linkedin.com/in/andreylomakin

**Comment by martingg88** [comment]
please give me command to export. thanks

**Comment by laa** [comment]
Fixed in 2.2.15 and develop, you can read about export/import there http://orientdb.com/docs/2.2/Export-and-Import.html
